core/field.js

```diff
@@ -154,6 +154,14 @@ Blockly.Field = function(value, opt_validator, opt_config) {
   opt_validator && this.setValidator(opt_validator);
 };
 
+/**
+ * Construct a Field from a JSON arg object.
+ * @param {!Object} _options A JSON object with options.
+ */
+Blockly.Field.fromJson = function(_options) {
+  throw Error("Field subclass doesn't define fromJson");
```

Comment:
Is there anyway we can tell people which field subclass is causing the problem?

Also style nit: single quotes and period in error msg. (I'm surprised lint doesn't catch the quotes).

Reply:
'this' is the field subclass, but there's no name I can find that's available to print.

I'm leaning towards deleting this function completely.  It will still throw, and the error would be more informative.  It's more of an architectural issue in that 'fromJson' is expected on all Fields, so Field should really define something.  This is a bit of an academic debate between a good design and a good implementation.  What do you think?

Regarding the string, it contains an apostrophe, thus the double-quotes.

Reply:
> I'm leaning towards deleting this function completely. It will still throw, and the error would be more informative. It's more of an architectural issue in that 'fromJson' is expected on all Fields, so Field should really define something. This is a bit of an academic debate between a good design and a good implementation. What do you think?

If you can get the type system to like it I'd say go ahead and delete it :D Otherwise this totally works, I just wanted to bring it up using a different message in case it was a possibility.

> Regarding the string, it contains an apostrophe, thus the double-quotes.

Oh whoops I totally missed that haha. I think the new styleguide recommends a template string in that case.

Reply:
Chris and I just had a 30 minute ultra-geek conversation about the options here.  The best code is no code.  Deleted.  :)

Reply:
> style nit: ... period in error msg

Oh, curious.  Is that a Blockly-specific convention?  I can't find mention of this in the Google styleguide.  I've usually avoided omitted them as the error message will normally be followed by "at filename:line:column…" when displayed.

Reply:
Oh nope that's just me being confused. I misread the diff and thought this PR added a period to the other error (actually removed it). I was going for consistency and instead caused confusion haha.